lib/puppet/pops/types/types.rb

```diff
@@ -840,11 +1021,26 @@ def generalize
       def hashed_elements
         @hashed ||= @elements.reduce({}) {|memo, e| memo[e.name] = e; memo }
       end
- 
+
       def hash
         @elements.hash
       end
 
+      def iterable?
+        true
+      end
+
+      def iterable_type
+        if self == DEFAULT
+          PIterableType.new(PTupleType.new([PAnyType::DEFAULT], PHashType::TUPLE_SIZE))
```

Comment:
Is this for the type `Struct[{}]` - If having one of those it does not really matter what the elementy type is, anything goes. I should use Tuple[Unit, Unit] I think (i.e. "whatever you want the type to be").